tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,511 @@
+
+    k4a_record_video_settings_t vid_settings = { 1920, 1080, 30 };
+    std::string k4a_cpp_ft_custom_vid_track = "K4A_CPP_FT_CUSTOM_VID_TRACK";
+    recorder.add_custom_video_track(k4a_cpp_ft_custom_vid_track.c_str(), "V_MPEG1", nullptr, 0, &vid_settings);
+
+    k4a_record_subtitle_settings_t st_track = { false };
+    std::string k4a_cpp_ft_custom_subtitle_track = "CUSTOM_K4A_SUBTITLE_TRACE";
+    recorder.add_custom_subtitle_track(k4a_cpp_ft_custom_subtitle_track.c_str(), "V_MPEG1", nullptr, 0, &st_track);
+
+    recorder.write_header();
+
+    for (int x = 0; x < 100; x++)
+    {
+        capture capture;
+        k4a_imu_sample_t imu;
+        if (kinect.get_capture(&capture, std::chrono::milliseconds(1000)))
+        {
+            recorder.write_capture(capture);
+        }
+
+        auto start = std::chrono::high_resolution_clock::now();
+        while (kinect.get_imu_sample(&imu, std::chrono::milliseconds(0)))
+        {
+            recorder.write_imu_sample(imu);
+            if (std::chrono::high_resolution_clock::now() - start < std::chrono::milliseconds(100))
+            {
+                break;
+            }
+        }
+        image color = capture.get_color_image();
+        recorder.write_custom_track_data(k4a_cpp_ft_custom_vid_track.c_str(),
+                                         color.get_device_timestamp(),
+                                         color.get_buffer(),
+                                         color.get_size());
+        color.reset();
+
+        image depth = capture.get_depth_image();
+        recorder.write_custom_track_data(k4a_cpp_ft_custom_subtitle_track.c_str(),
+                                         depth.get_device_timestamp(),
+                                         depth.get_buffer(),
+                                         depth.get_size());
+        capture.reset();
+    }
+    recorder.flush();
+    kinect.stop_cameras();
+    kinect.stop_imu();
+}
+
+static void test_playback(void)
+{
+    playback pb = playback::open(MKV_FILE_NAME.c_str());
+    ASSERT_TRUE(pb); // bool operation
+
+    {
+        playback pb_missing_file;
+        ASSERT_THROW(pb_missing_file = playback::open("./This_file_is_not_here.mkv"), error);
+
+        playback pb_empty;
+        ASSERT_TRUE(pb);               // bool operation
+        ASSERT_FALSE(pb_empty);        // bool operation
+        ASSERT_FALSE(pb_missing_file); // bool operation
+        ASSERT_TRUE(pb != pb_empty);
+        ASSERT_FALSE(pb == pb_empty);
+    }
+
+    {
+        // playback pb_empty = pb; // deleted operation
+    }
+
+    playback pback;
+    ASSERT_TRUE(pb);     // bool operation
+    ASSERT_FALSE(pback); // bool operation
+    pback = std::move(pb);
+    ASSERT_FALSE(pb);   // bool operation
+    ASSERT_TRUE(pback); // bool operation
+
+    std::vector<uint8_t> raw_cal = pback.get_raw_calibration();
+    std::cout << "calibration is : ";
+    for (const uint8_t data_char : raw_cal)
```

Comment:
>d [](start = 23, length = 1)

nit: you're looping over these by value, not by reference, so it's actually copying each char into a temporary here.  This is test code so it probably doesn't really matter, but you probably want an & here